frontend/src/lib/utils/PublishSubscribeDelegate.ts

```diff
@@ -38,6 +38,16 @@ export class PublishSubscribeDelegate<TTopicPayloads extends TopicPayloads> {
         }
     }
 
+    subscribe(topic: keyof TTopicPayloads, handler: () => void): () => void {
+        const subscribers = this._subscribers.get(topic) || new Set();
+        subscribers.add(handler);
+        this._subscribers.set(topic, subscribers);
+
+        return () => {
+            subscribers.delete(handler);
+        };
+    }
```

Comment:
I think you can make `makeSubscriberFunction` below to use your new `subscribe` function.


```
makeSubscriberFunction(topic: keyof TTopicPayloads): (onStoreChangeCallback: () => void) => () => void {
        // Using arrow function in order to keep "this" in context
        const subscriber = (onStoreChangeCallback: () => void) => {
            return this.subscribe(topic, onStoreChangeCallback);
        };

        return subscriber;
    }
```